here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/query/WhereClauseBuilder.kt

```diff
@@ -160,6 +163,81 @@ class WhereClauseBuilder(private val request: ReadFeatures) {
         }
     }
 
+    private fun whereTags() {
+        val tagQuery = request.query.tags
+        if(tagQuery != null){
+            if (where.isNotEmpty()) {
+                where.append(" AND (")
+            } else {
+                where.append(" (")
+            }
+            whereNestedTags(tagQuery)
+            where.append(")")
+        }
+    }
+
+    private fun whereNestedTags(tagQuery: ITagQuery) {
+        when (tagQuery) {
+            is TagNot -> not(tagQuery.query, this::whereNestedTags)
+            is TagOr -> or(tagQuery.filterNotNull(), this::whereNestedTags)
+            is TagAnd -> and(tagQuery.filterNotNull(), this::whereNestedTags)
+            is TagQuery -> resolveSingleTagQuery(tagQuery)
+        }
+    }
+
+    private fun resolveSingleTagQuery(tagQuery: TagQuery) {
+        when (tagQuery) {
+            is TagExists -> {
+                where.append("$tagsAsJsonb ?? '${tagQuery.name}'")
+            }
+
+            is TagValueIsNull -> {
+                where.append("${tagValue(tagQuery)} = null")
```

Comment:
That does not work, it need to be `something IS NULL`, NULL is like NaN, any value compared against it will be _false_ (well, not exactly, but it is good to remember it like this).

Reply:
👍  fixed